packages/mobile/src/identity/commentEncryption.ts

```diff
@@ -221,42 +223,52 @@ function* verifyIdentityMetadata(data: IdentityMetadataInTx[]) {
   // 2. Some duped phone hashes could be true, others false
   // 3. And a given phoneHash could legitimately have multiple addresses
 
+  const contractKit = yield call(getContractKit)
+  const accountsWrapper: AccountsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAccounts,
+  ])
+
   if (!data || !data.length) {
     return []
   }
 
-  const phoneHashes = new Set<string>()
-  data.map((d) => {
-    const phoneHash = getPhoneHash(d.e164Number, d.salt)
-    phoneHashes.add(phoneHash)
-    d.phoneHash = phoneHash
-  })
-
-  const lookupResult: IdentifierLookupResult = yield call(
-    lookupAttestationIdentifiers,
-    Array.from(phoneHashes)
-  )
-
-  return data.filter((d) => {
-    const onChainAddresses = getAddressesFromLookupResult(lookupResult, d.phoneHash!)
+  const verifiedTx = []
+  for (const metadata of data) {
+    const phoneHash = getPhoneHash(metadata.e164Number, metadata.salt)
+    metadata.phoneHash = phoneHash
+    const accountAddresses: Address[] = yield call(lookupAccountAddressesForIdentifier, phoneHash)
+
+    // Check that there are verified addresses.
+    const onChainAddresses: Address[] = yield call(
```

Comment:
Nit: Can we rename this to `verifiedAccountAddresses`? I don't think `onChainAddresses` means anything in specific